### PR TITLE
add support to uiox as input string to opaeuio, read8/16, write8/16 t…

### PR DESCRIPTION
**Pull request by anandaravuri**
…o pyopaeuio

  - add support to uiox  input string to opaeuio open function users can pass dfl_dev.x or uiox to open function

   - add read8/16, write8/16, get size function to pyopaeuio

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: libraries/libopaeuio/opaeuio.c [hunk 1]
```diff
@@ -263,71 +265,78 @@ STATIC int opae_uio_init(struct opae_uio *u, const char *dfl_device)
...
-			     glob_fmts[i], dfl_device) < 0) {
-			ERR("snprintf() failed\n");
-			return 1;
-		}
-
-		glob_res = opae_glob(path_expr, GLOB_NOSORT, NULL, &globbuf);
-		if (glob_res || !globbuf.gl_pathc) {
-			opae_globfree(&globbuf);
-			continue;
+	// Check for uio sting in dfl_device
```
Comment (tswhison) [inline libraries/libopaeuio/opaeuio.c:268]:
```suggestion
	// Check for "/dev/uio" string in dfl_device
```

File: libraries/libopaeuio/opaeuio.c [hunk 2]
```diff
@@ -263,71 +265,78 @@ STATIC int opae_uio_init(struct opae_uio *u, const char *dfl_device)
...
+	ptr = strstr(dfl_device, "uio");
```
Comment (tswhison) [inline libraries/libopaeuio/opaeuio.c:269]:
```suggestion
	ptr = strstr(dfl_device, "/dev/uio");
```

File: libraries/libopaeuio/opaeuio.c [hunk 3]
```diff
@@ -263,71 +265,78 @@ STATIC int opae_uio_init(struct opae_uio *u, const char *dfl_device)
...
+	if (ptr == NULL) {
```
Comment (tswhison) [inline libraries/libopaeuio/opaeuio.c:270]:
```suggestion
	if (ptr != dfl_device) {
```

File: libraries/libopaeuio/opaeuio.c [hunk 4]
```diff
@@ -338,7 +353,9 @@ STATIC int opae_uio_init(struct opae_uio *u, const char *dfl_device)
 out_destroy:
 	opae_uio_destroy(u);
 out_glob_free:
-	opae_globfree(&globbuf);
+	if (glob_res || !globbuf.gl_pathc) {
```
Comment (tswhison) [inline libraries/libopaeuio/opaeuio.c:356]: Are these checks necessary / correct?

Comment (anandaravuri) [reply libraries/libopaeuio/opaeuio.c:356]: if user pass uio, with out this checks sig fault

if user pass dfl_dev.0 , no sig fault

File: libraries/libopaeuio/opaeuio.c (hunk 3 above)
Comment (anandaravuri) [reply libraries/libopaeuio/opaeuio.c]: fixed

File: libraries/libopaeuio/opaeuio.c (hunk 1 above)
Comment (anandaravuri) [reply libraries/libopaeuio/opaeuio.c:268]: fixed

**Review by tswhison — APPROVED** [review]

**Pull request merged**